api/views/pois.py

```diff
@@ -0,0 +1,104 @@
+from api import app, db
+from api.models import POI, Media, Link, Story, StoryPOI
+from api.utils import create_response, InvalidUsage
+from sqlalchemy.sql.expression import func
+from flask import Blueprint, request, jsonify
+import json
+from datetime import date
+from dateutil.parser import parse
+
+mod = Blueprint('POIS', __name__)
+
+def poi_links_media_stories(poi_dict):
+    """
+    Gets additional links, media, and stories for a POI
+    """
+    poi_links = Link.query.filter(Link.poi_id == poi_dict['_id'])
+    poi_media = Media.query.filter(Media.poi_id == poi_dict['_id'])
+    poi_stories = Story.query.join(StoryPOI, Story.id == StoryPOI.story_id).filter(StoryPOI.poi_id == poi_dict['_id'])
+    poi_dict['links'] = [j.to_dict() for j in poi_links]
+    poi_dict['media'] = [k.to_dict() for k in poi_media]
+    poi_dict['stories'] = [l.to_dict() for l in poi_stories]
+    return poi_dict
+
+@app.route('/pois', methods=['GET'])
+def pois_get_by_map_year_or_story():
+    map_year, story_id = request.args.get('map_year'), request.args.get('story_id')
+    if map_year is not None and story_id is None:
+        # Find POIs by map_year
+        pois = POI.query.filter(POI.map_year == int(map_year))
+        if pois.count() == 0:
+            return create_response(status=404, message='No POIs found')
+    elif story_id is not None and map_year is None:
+        # Find POIs by story_id
+        pois = POI.query.join(StoryPOI, POI.id == StoryPOI.poi_id).filter(StoryPOI.story_id == int(story_id))
+        if pois.count() == 0:
+            return create_response(status=404, message='No POIs found')

```

Comment:
since this `if` block is the same as above, we can move it outside the block to make it more dry